### DISCUSSION TITLE
Milvus支持在集群上完成索引建立吗？

### DISCUSSION THREAD

**Discussion by rangehow** [milvus-io/milvus#36476] [Q&A and General discussion] [answered]
我比较熟悉FAISS，现在因为预训练需求，我需要将数据量变得非常大，因此我在寻求一个支持在集群/多节点上完成索引、查询的开源数据库。Milvus是满足这个需求的数据库吗？非常感谢伟大的社区人员们~

**Comment by yhmo** [comment, marked as answer]
“在集群/多节点上完成索引、查询的开源数据库“ ---------- 这就是milvus，它把数据分成很多分片，不同的分片可以在不同的节点上建索引，不同的节点上查询。每个分片有单独的索引，多个分片的查询结果合并成一个最终结果返回给客户端。

**Reply by rangehow** [reply]
非常感谢您的回答，我还想再具体请教一下。这意味着，每个节点上的**索引**和**持有的数据**都是不同的吗，这个行为能确保返回结果和单节点的返回结果是一致的吗？

**Reply by rangehow** [reply]
如果最终能保证是一致的，那就太好了，milvus就是我所需要的答案。如果有可能不一致的话，潜在的性能下滑是多少呢。

**Reply by xiaofan-luan** [reply]
> 非常感谢您的回答，我还想再具体请教一下。这意味着，每个节点上的**索引**和**持有的数据**都是不同的吗，这个行为能确保返回结果和单节点的返回结果是一致的吗？

不能完全保证查询结果是一致的。因为做了分片，index 的结构有变化，但是99%以上的情况都是一致的。

**Reply by yhmo** [reply]
milvus内部节点间通过time-tick检查数据的一致性: https://milvus.io/docs/time_sync.md#Time-synchronization-system-timetick
查询时，可以指定一个consistency_level参数来确定该次查询想要什么级别的一致性:
https://milvus.io/docs/consistency.md#Consistency

设为Strong就会等待所有节点都同步到最新数据之后再执行查询。
设为Bounded允许在一定的时间窗口内的数据不一致。
设为Eventually不管一致性如何都立即执行查询。

**Reply by rangehow** [reply]
> 不能完全保证查询结果是一致的。因为做了分片，index 的结构有变化，但是99%以上的情况都是一致的。

这个一致性还是非常棒的，谢谢！: )

**Reply by rangehow** [reply]
> milvus内部节点间通过time-tick检查数据的一致性: https://milvus.io/docs/time_sync.md#Time-synchronization-system-timetick 查询时，可以指定一个consistency_level参数来确定该次查询想要什么级别的一致性: https://milvus.io/docs/consistency.md#Consistency
> 
> 设为Strong就会等待所有节点都同步到最新数据之后再执行查询。 设为Bounded允许在一定的时间窗口内的数据不一致。 设为Eventually不管一致性如何都立即执行查询。

我学习一下，感谢您提供的材料~
